packages/webdriver-utils/src/providers/genericProvider.js

```diff
@@ -191,16 +203,27 @@ export default class GenericProvider {
     ];
   }
 
+  updateYFactor(location) {
+    if (this.currentOs === 'iOS') {
+      if (location.y === 0) {
+        this.pageYShiftFactor += (-this.scrollYFactor);
+      }
+    }
+  }
+
   async getRegionObject(selector, elementId) {
-    const scaleFactor = parseInt(await this.metaData.devicePixelRatio());
+    const scaleFactor = await this.metaData.devicePixelRatio();
     const rect = await this.driver.rect(elementId);
     const location = { x: rect.x, y: rect.y };
     const size = { height: rect.height, width: rect.width };
+    // Update YFactor Element is not visible in viewport
+    // In case of iOS if the element is not visible in viewport it gives 0,0 as coordinate.
+    this.updateYFactor(location);
```

Comment:
@nilshah98  It was important to get the functionality getting tested. Separating the function makes the function readability look better imo  and testing becomes much easier for the various cases. So tomorrow if we find a new behaviour it can be handled in this function.

Reply:
This is further simplified to a simple function and logic is shifted to single function.